### DISCUSSION TITLE
SickGear and old ARM TS120

### DISCUSSION THREAD

**Discussion by DynaMight1124** [OneCDOnly/sherpa#289] [General]
Hi

I look after a friends old TS120, which is pretty much used for TV Shows. He's been running SickGear for the past few years on it without many issues but just suffered major disk failure. He's ordered a new drive and I'll do a fresh install however I've noticed that support for SickGear has been dropped.

I've grabbed the last 'compatible' qpkg 240108 but will this even work in a fresh install or is he out of luck? The compatibility notes mention about a PyPi module not compiling. 

Thanks

**Comment by OneCDOnly** [comment]
Hi mate. :)

Ah, good question. I've not tried this since the Entware packages were updated a short while ago. I saw there are now some Rust-compiled modules available, which had been an issue for the older hardware. I'll need to re-test and see if it's now possible. 

I'll add this to my todo list, but can't say when when I'll get to test this though.

**Reply by DynaMight1124** [reply]
Thank you, I'll give the old qpkg a go in the meantime when the new drive arrives and see what happens.

**Reply by DynaMight1124** [reply]
I did try to install the old qpkg but I got error: [OSickGear] unable to install addons: virtual environment does not exist! so I assume its a no go.

This did allow me to try Medusa, another Sickboard fork. It installed great and ran great (I'm not really sure which one is lightest in terms to resources) however its not happy when I reboot, it wont start back up again, I left it 15-30mins but still nothing (even left it overnight and still didnt start) seemingly a stop/start from within the WebUI doesnt work either but if I restart from SSH it will restart, so far taken between 10-30mins to restart but has worked every time.

I actually think SickGear suffered the same issue as I had to remotely log into my friends a few times and SSH usually seemed to fix it. He's not very good with tech :)

I know its an old box so possibly something related to speeds, maybe something like Python hadnt fully started. I'm not sure if theres anything you can do, I've suggested to my friend to donate it to the museum :) 

I feel pretty bad for keep bugging you about an old box, please tell me if I'm pushing my luck, I honestly dont mind just putting up with what we got.

P.S. I'm not sure if theres a bug on the latest release regarding Deluge, happened on my box and this box now too. If I run a upgrade all, it installs Deluge Server even though it was never previously installed on either.

**Reply by DynaMight1124** [reply]
Hopefully helps: 
https://pastebin.com/wghaZiTh <-- sherpa restart log from SSH verbose
https://pastebin.com/TZKgWcjU <-- WebGUI stop/start from var/log/OMedusa.log

**Reply by OneCDOnly** [reply]
Good work! 🤓 

Thanks for posting your logs, that helps move thing along. I'll need to investigate this further. I'll try to do-so this weekend. 

That does sound like a bug when upgrading all installs Deluge-server. I'll need to check that out too. 

Cheers mate. 👍🏽

**Comment by OneCDOnly** [comment]
Damn, I hadn't noticed that the new Entware package updates no-longer apply to armv5 NAS models. 😞 

https://entware.net/2024/02/23/Changelog.html

This means all armv5 NAS are stuck with their existing IPK versions and those won't be updated anymore. 

Which means the updates I had hoped would allow us to install Medusa and Sickgear on armv5 NAS (like the TS-120) don't apply. 

Sorry mate, I think it's the end-of-line for that old NAS.

**Reply by DynaMight1124** [reply]
Ah damn, thats a shame. Where does that leave us, Medusa is currently running ok, just worried that when he reboots at some stage in the future its gonna try to update/try to update and break the whole thing without any way of getting any sickbeard ports back running again. 

Obviously I totally understand if its just one of those things, just wondering if theres anything that can help it run for a little longer. Just spent the last few days fine tuning it to the new drive and its running very nice for a vintage NAS :)

**Reply by OneCDOnly** [reply]
You might try disabling the auto-update feature of the service-script (this is something only sherpa service-scripts can do). That should shorten the startup time a bit:
```
/etc/init.d/omedusa.sh disable-auto-update
```
That setting will be saved, so then you can start/restart Medusa as usual.

**Reply by OneCDOnly** [reply]
BTW: I'm doing a lot of testing this morning, and can confirm Medusa installs fine on my TS-220. It's stuck on the old Python version (3.11.4), but seems to be launching and running OK. 👍🏽

**Reply by OneCDOnly** [reply]
Hmm, interesting development. I just got Sickgear to install on my TS-220. I had to tweak the Python deps a little, but it works. 🤓

**Comment by DynaMight1124** [comment]
Ok cool, I might stop updates all together. Just worried a future update (could be next week or a year away!) could stop it working all together with no easy way to get it working again. So maybe safer to stop future updates and ride it out.

Yeah Medusa is actually running pretty good on his TS120, takes some time to start (and wont start when rebooted but I've created a batch script to restart via ssh for when he does reboot) but once its started it runs pretty smoothly.

You reckon ARM64 is a fairly safe bet for a future supported QNAP nas? I saw a fairly cheap TS128A and that was still on the latest firmware so feels like it would remain supported by Entware/Python etc for a while yet.

Just saw you message about Sickgear too, would be awesome if it could work as a backup. potentially leave one updating and the other frozen.

**Comment by OneCDOnly** [comment]
> Ok cool, I might stop updates all together. Just worried a future update (could be next week or a year away!) could stop it working all together with no easy way to get it working again. So maybe safer to stop future updates and ride it out.

Agree. 👍🏽 

> Yeah Medusa is actually running pretty good on his TS120, takes some time to start (and wont start when rebooted but I've created a batch script to restart via ssh for when he does reboot) but once its started it runs pretty smoothly.

Have you also installed the IncreaseTimeouts QPKG? It should help on the older NAS. 

> You reckon ARM64 is a fairly safe bet for a future supported QNAP nas?

Yes, I suspect arm64 (aarch64) will be around for a while. 

>  I saw a fairly cheap TS128A and that was still on the latest firmware so feels like it would remain supported by Entware/Python etc for a while yet.

The TS-128A was released over 6 years ago, and is a little-low on RAM (only 1GB). You might try for something more recent like a TS-233 or TS-133. These are shipped with 2GB shared RAM, which means you have about 1.5GB RAM available to QTS, the other 512MB is reserved for transcoding use-only. 

> Just saw you message about Sickgear too, would be awesome if it could work as a backup. potentially leave one updating and the other frozen.

Yup, I'm still working on it, and I'm trying to make Sickgear and Medusa available on the widest possible range of QNAP models. 🤓

**Reply by DynaMight1124** [reply]
I'll keep the TSx33 in mind, trying to work out which QNAP is which is a bit of a mine field! I reckon 1GB would probably be ok for him if pushed, its currently not too bad with 512MB although I'd imagine the newer firmware/packages have a little bit more bloat so extra maybe handy!

Funny thing is I did see the IncreaseTimeouts QPKG mentioned when I scanned the logs, I looked in your Github in the QPKG section and didnt see it, Google searched too but obviously not good enough :) I found it in the proper Wiki Packages! 

I've disabled the auto updates and installed IncreaseTimeouts and its done the trick, Medusa starts when rebooting, it stops/starts in QTS. I tested Sherpa restart too and it took 1m23sec to restart, compared to 8mins plus. Sounds like the timeout was 3mins before so assuming thats per QPKG then may had got away without it by just disabling auto updates but either way, super happy it starts when rebooting. Thank you very much.

**Comment by OneCDOnly** [comment]
BTW: have you tried Clinton Hall's "official" Sickgear QPKG? https://github.com/SickGear/SickGear/wiki/Install-SickGear-[5]-QNAP

**Reply by DynaMight1124** [reply]
I am pretty sure I did, (I know of his QNAP work as I pretty much ran all his QPKG's but I think he stepped back a few years back so I moved on), but I do remember trying one of his SickGear packages, possibly when I first set it up for him and at the time it wasnt compatible with the older CPU in the TS120, which is why I was happy you supported all types and never looked back.

I have just tried it and it installed without error and runs. It shows just how much snappier the Medusa interface is compared to SG! Thats good news for my friend though, I'll leave it installed but disabled, I'll aim to install your version of SG for a triple threat! 

Edit: Saying that, it booted fine when first installed but didnt come back after a stop/start in QTS but at least it showed some signs of life :)
